crates/compute/src/layer.rs

```diff
@@ -122,6 +125,16 @@ pub trait ComputeLayer<F> {
 		coordinates: &[F],
 		data: &mut <Self::DevMem as ComputeMemory<F>>::FSliceMut<'_>,
 	) -> Result<(), Error>;
+
+	fn fold_left<'a>(
```

Comment:
Missing docstring, and I prefer the parameter names [in the original branch](https://github.com/IrreducibleOSS/binius/blob/jimpo/2025-03-25-hal-playtime/crates/hal/src/v2.rs#L177). Is there a reason you added parameters for the sizes rather than inspecting the slice lengths?

Reply:
The log len of `query` can be inferred from the slice len now that the tower level is guaranteed to be log2(128).

Regarding parameter names, I matched those in the Binius definition at https://github.com/IrreducibleOSS/binius/blob/main/crates/math/src/fold.rs#L132. Which names do yo prefer?

Reply:
I prefer the names I chose for the API in the linked branch. IMO they reflect what the operation does better, whereas the `binius_math::fold` parameter names match how it is used in the protocol code.

Reply:
Please import the parameter names and docstring from https://github.com/IrreducibleOSS/binius/blob/jimpo/2025-03-25-hal-playtime/crates/hal/src/v2.rs#L177